### PR TITLE
VMs start running even if the db url/credentials is incorrect or db i…

### PR DESCRIPTION
**Pull request by qibobo**
…s unreachable for servicebroker

[#154807154]

### CODE REVIEW

**Comment by cdlliuy** [comment]
LGTM

File: servicebroker/lib/index.js [hunk 1]
```diff
@@ -23,5 +23,10 @@ if (validateResult.valid === false) {
     logger.error("Invalid configuration: " + validateResult.message);
     throw new Error('settings.json is invalid');
 }
-
-var server = require(path.join(__dirname, '../lib/server.js'))(settings, catalog);
+var dbErrorCallback = function(err){
+        if(err){
+            logger.error('DB configuration is incorrect, server will exit', err);
+            process.exit(1);
```
Comment (cdlliuy) [inline servicebroker/lib/index.js]: similar to api server, should we use a consistent way to handle FATAL errors?

File: servicebroker/test/server_test.js [hunk 2]
```diff
@@ -72,3 +74,73 @@ describe("Auth for RESTful API", function() {
...
+
+  beforeEach(function() {
+    settings = require(path.join(__dirname, '../lib/config/setting.js'))((JSON.parse(
+  fs.readFileSync(configFilePath, 'utf8'))));
+  });
+
+  context("Wrong db configuration", function() {
+    var server;
+    afterEach(function(done) {
+      server.close(done);
```
Comment (cdlliuy) [inline servicebroker/test/server_test.js]: could  we check whether the server is created before close it?

**Comment by cdlliuy** [comment]
LGTM

**Comment by boyang9527** [comment]
LGTM

**Pull request merged**